### PR TITLE
Dose calculation

### PR DESCRIPTION
**Pull request by juhor00**
Implemented dose calculation using two builder patterns.
`DiagnoseResponseBuilder` builds `DiagnoseResponse` object, which contains basic information and the treatment options (primary, secondary or penicillin allergic). They are ordered based on primary / secondary value.
`AntibioticTreatmentBuilder` builds `AntibioticTreatment` object, which contains all needed information about the results, formula and instructions. The results are calculated based on weight.

I made an assumption regarding the selection of antibiotics, I will describe it in specifics via a code comment.

If the diagnosis has no antibiotic treatment, a `NoAntibioticTreatmentException` is thrown and API returns `NoAntibioticTreatment` object, which is not yet implemented.

### CODE REVIEW

**Comment by juhor00** [comment]
Sori, tää kasvo ihan liian isoks muutokseks yhdelle kerralle. Oli alunperin tarkotus tehdä pienemmissä palasissa, mutta useat vaatimusten muutokset keskenkaiken sai tän paisuun. Toivottavasti jossain määrin pystytte kattoon tän läpi :D

File: backend/antibiootit/src/main/java/fi/tuni/koodimankelit/antibiootit/builder/DiagnoseResponseBuilder.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        }
+    }
+
+    
+    /** 
+     * Returns suitable antibiotic choise from given treatment based on weight
+     * @param treatment specific treatment
+     * @return Antibiotic preferred antibiotic
+     */
+    private Antibiotic getSuitableAntibiotic(Treatment treatment) {
```
Comment (juhor00) [inline backend/antibiootit/src/main/java/fi/tuni/koodimankelit/antibiootit/builder/DiagnoseResponseBuilder.java:142]: Here's the assumption I mentioned. A treatment can have multiple different antibiotics, and those can have multiple different strengths. These strengths have their minimum weights. For example these could be:

Antibiotic A
- minWeight 0

Antibiotic B
- minWeight 40
- minWeight 60

This function sorts minWeight to descending order within an antibiotic. Then it compares the antibiotics, which one has the higher minWeight is first. Then the function iterates the antibiotics and their strengths and chooses the strength with highest applicable minWeight.

This assumes, that different antibiotics don't have overlapping minWeight strengths. For example:

Antibiotic C
- minWeight 0
- minWeight 50

Antibiotic D
- minWeight 40
- MinWeight 60

They would be ordered:
60 - D
40 - D
50 - C
0   - C

If our patient is 55kg, the 40 - D is chosen, even though 50 - C is the most suitable one. I think the antibiotics' strength's minimum weights don't overlap like this?

**Review by pampula — APPROVED** [review]
You did a lot of work, great job! The code looks fine and professional. The minWeight issue indeed needs to be confirmed with the customer, but this solution can be merged to main for now in my opinion.

**Review by KiviluomaNeera — APPROVED** [review]
Everything looks good!

**Pull request merged**